CMakeLists.txt

```diff
@@ -0,0 +1,179 @@
+cmake_minimum_required(VERSION 3.25...3.29)
+
+#[=============================================================================[
+#                           Basic project definition                           #
+]=============================================================================]
+
+list(APPEND CMAKE_MESSAGE_CONTEXT Wannier90)
+project(Wannier90
+		VERSION 4.0.0
+		DESCRIPTION "Compute maximally-localised Wannier functions."
+		HOMEPAGE_URL https://www.wannier90.org
+		LANGUAGES Fortran C
+		)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS OFF)
```

Comment:
There are some new C sources, but header only? Probably we don't need to add C to the main project in that case.

Reply:
Adding `C` is useful for compiling with MKL, without this cmake cannot detect MKL automatically
https://cmake.org/cmake/help/latest/module/FindBLAS.html#intel-mkl

It is a bit annoying since we are actually a fortran code, but without this users need to define a lot of cmake variables for MKL.

Reply:
What about using `find_package(MKL)` by default if it's intel compiler, otherwise have an option `WANNIER90_MKL` option that uses either MKL or default BLAS/LAPACK? I am not sure if it works on a pure Fortran project, but we can try and see

Reply:
Did a test, we can add some `if` conditionals to find either MKL or BLAS/LAPACK, however, in the
```
target_link_libraries(Wannier90_lib PRIVATE
        BLAS::BLAS LAPACK::LAPACK)
```
I guess we need to add another `if` for MKL as well, not sure if there is a simpler solution.

On the other hand, by enabling `C` language everything works automatically.

Reply:
Ok, I suggest we postpone the decision of using either implementation, because @giovannipizzi warned me that there is an issue between LAPACK and macos, which we need to think how to handle.